pkg/controller/networkaddonsconfig/networkaddonsconfig_controller.go

```diff
@@ -498,6 +495,65 @@ func (r *ReconcileNetworkAddonsConfig) stopTrackingObjects() {
 	r.statusManager.SetFromPods()
 }
 
+func updateObjectsLabels(crLables map[string]string, objs []*unstructured.Unstructured) error {
+	var err error
+	for _, obj := range objs {
+		labels := obj.GetLabels()
+		if labels == nil {
+			labels = map[string]string{}
+		}
+		// Label objects with version of the operator they were created by
+		labels[cnaov1.SchemeGroupVersion.Group+"/version"] = operatorVersionLabel
+
+		labels[names.COMPONENT_LABEL_KEY] = names.COMPONENT_LABEL_DEFAULT_VALUE
```

Comment:
I think we should move these 2 lines that initialize the `managed-by` and `component` labels to be inside `updateLabelsFromCR`. It would make it more visual that we're setting a default value then rewriting it, and make it harder for future us to mistakenly reorder the commands without noticing we're changing the label behavior.

Reply:
we don't override `component` so i will just remove the `managed-by`

Reply:
Thinking about it, I think we should also override `component`, so both should be moved to `updateLabelsFromCR`

Reply:
isnt moving it to `updateLabelsFromCR`
breaks the behavior that we are setting a default value which is not from the CR ?

Reply:
perhaps then `initializeAndUpdateLabelsFromCR`

Reply:
Prepare to not over complicate the function name